### PR TITLE
Added trust anchor rfc

### PR DESCRIPTION
**Pull request by vedangj044**
Added RFC for Trust Anchor feature in DRG.

### CODE REVIEW

File: active/005-trust-anchor-management.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+- drg `trust` ...
+- drg `trust-anchor` ...
+- drg `adm` ...
+- your suggestions
+
+Let's go with `trust` for this document.
+
+Next up is the functionality this command adds to drg. We have a `device` entity that is contained in an isolated box called `app`. This app would have its private key, the device generates a certificate signing request, which is signed by the private key. From that point, this device can authenticate using the certificate. This app works as the trust anchor of the device.
+
+`drg trust create app <app_name>`
```
Comment (lulf) [inline active/005-trust-anchor-management.md:22]: How about `drg trust create --app <app_name>` . I think it's easy to be confused on what you're actually creating vs. passing arguments on what you're going to trust.

Comment (vedangj044) [reply active/005-trust-anchor-management.md:22]: Right :+1:

File: active/005-trust-anchor-management.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+`drg trust revoke app <app_name>` This command can be used to delete a trust anchor, the user would be prompted with a list of available CA ( if more than one ).
+
+- `--cert <hash_of_cert>` - Users can specify the CA to delete here.
+- `--y` - Users can add this flag, to skip the confirmation prompt.
+
+All devices using this certificate would be marked as unauthenticated. As this is a critical operation, users should be prompted with a cryptographic challenge, maybe a random hash with they have to sign using the private key, thus allowing certificate revocation on the successful completion of the challenge.
+
+## Implementation
+
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:52]: todo : find a rust library to do the cert work that uses `ring` if possible. the openSSL bindings rust crate is a nightmare to cross-compile :) And i'd like to keep `drg` pure rust :)

File: active/005-trust-anchor-management.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- `--file <key_file>` - Users can use their own private keys.
+- `--algo <algorithm>` - Users can fine tune the certificate generation by choosing the algorithm.
+- `--org <organization_name> --cm <comman_name>` - Users can specify the certificate information using flags, or would be prompted to enter it.
+
+We can define a context in drg, this can be useful to decide on the default algorithm to generate the certificates and keys.
+
+`drg trust create device <device_name> --app <app_name> --out <file_path>`
+This command requires the private key of the app and signs a new certificate and outputs the file to the specified file path, or it uploads it to the cloud with some encryption. The certificate can be later flashed to the device.
+
+- `--file <csr_file>` - Users can use their own CSR.
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:34]:
```suggestion
- `--csr <csr_file>` - Users can use their own CSR.
```

File: active/005-trust-anchor-management.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+`drg trust create all --app <app_name>`
+As the number of devices increases, this process becomes tiresome, so with this command certs for all the devices in this app would be generated and stored in the specification. This requires the private key.
+
+`drg trust export device <device_name> --app <app_id> <file_name>`
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:39]:
```suggestion
`drg trust download device <device_name> --app <app_id> <file_name>`
```

would be more explicit IMHO

File: active/005-trust-anchor-management.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+This command downloads the device certificates from the cloud and stores them locally. Probably, this happens in the flashing process.
+
+`drg trust print app <app_name>` This command displays the list of all the thumbprints ( digest hash ) of all the root CAs.
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:42]:
```suggestion
`drg trust get app <app_name>` This command displays the list of all the thumbprints ( digest hash ) of all the root CAs.
```
to keep some consitency with other drg subcommands

File: active/005-trust-anchor-management.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+`drg trust revoke app <app_name>` This command can be used to delete a trust anchor, the user would be prompted with a list of available CA ( if more than one ).
+
+- `--cert <hash_of_cert>` - Users can specify the CA to delete here.
+- `--y` - Users can add this flag, to skip the confirmation prompt.
+
+All devices using this certificate would be marked as unauthenticated. As this is a critical operation, users should be prompted with a cryptographic challenge, maybe a random hash with they have to sign using the private key, thus allowing certificate revocation on the successful completion of the challenge.
+
+## Implementation
+
+TODO: Figuring out where to store the private keys.
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:53]: XDG_CACHE could be appropriate

**Comment by jbtrystram** [comment]
There a couple of points that needs some more design before starting to implement things : 

 - Spec out the payload for the cert in the `app` spec section. 
 - Same as above but for the device spec section
 - If we allow pushing device private key and cert to the cloud it MUST be encrypted, so there are some metadata going along, the payload needs specification as well

File: active/005-trust-anchor-management.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+The specification of each device needs to contain a unique mapping to its physical counterpart, this can be accomplished by storing the MAC Address + Service number of every device or storing the MAC address as a JWT token provided by the server.
+
+The generation of the private keys and certificates would be done using rust-openssl.
```
Comment (jbtrystram) [inline active/005-trust-anchor-management.md:57]: This one is a pain to cross compile. Is there a pure rust crate that could do it ? Maybe look into `ring` and `webpki`
https://github.com/briansmith/webpki

**Pull request merged**